src/components/common/coordsValidate.ts

```diff
@@ -1,5 +1,46 @@
 export function parseCoords(coord): string[] {
   const regExp = /^(\-?\d+(?:\.\d+)?),\s*(\-?\d+(?:\.\d+)?)$/;
   const result = regExp.exec(coord);
-  return result !== null ? result.slice(1) : null;
+  if (result !== null) {
+    return result.slice(1);
+  } else {
+    //it's geopoint/geostring type
+    let geoObject;
+    try {
+      geoObject = JSON.parse(coord);
+    } catch (e) {
+      throw new Error("Invalid GeoJSON object.");
+    }
```

Comment:
Maybe we should allow the original error to throw here, to let the programmer understand the cause of the error, that in this case would be a parsing error.

Reply:
Done. Addresed in 6aa1ac5